src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,139 @@
+import { getPeople } from '../../api';
+import { Loader } from '../Loader';
+import { PeopleFilters } from '../PeopleFilters';
+import { FilterParams } from '../../types/FilterParams';
+import { SortColumns } from '../../utils/SortColumns';
+import { SortOrders } from '../../utils/SortOrders';
+import { sortPeopleBy } from '../../utils/functions';
+
+function filterPeople(people: Person[], {
+  queryFilter,
+  centuryFilter,
+  sexFilter,
+  sortFilter,
+  sortOrder,
+}: FilterParams) {
+  let peopleCopy = [...people];
+
+  if (queryFilter) {
+    const normalizedQuery = queryFilter.toLowerCase();
+
+    peopleCopy = peopleCopy.filter(person => {
+      const normalizedName = person.name.toLowerCase();
+      const normalizedMother = person.motherName?.toLowerCase() || '';
+      const normalizedFather = person.fatherName?.toLowerCase() || '';
+
+      return normalizedName.includes(normalizedQuery)
+        || normalizedMother.includes(normalizedQuery)
+        || normalizedFather.includes(normalizedQuery);
+    });
+  }
+
+  if (centuryFilter && centuryFilter.length) {
+    peopleCopy = peopleCopy.filter(person => {
+      const currentCentury = Math.floor(person.born / 100) + 1;
+
+      return centuryFilter.includes(currentCentury.toString());
+    });
+  }
+
+  if (sexFilter) {
+    peopleCopy = peopleCopy.filter(person => {
+      return person.sex === sexFilter?.toString();
+    });
+  }
+
+  if (sortFilter && sortOrder) {
+    peopleCopy = sortPeopleBy(
+      peopleCopy,
+      sortFilter as SortColumns,
+      sortOrder as SortOrders,
+    );
+  }
+
+  return peopleCopy;
+}
+
+export const PeoplePage: React.FC = () => {
+  const [searchParams] = useSearchParams();
+
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const visiblePeople = useMemo(() => {
+    const filters: FilterParams = {
+      queryFilter: searchParams.get('query'),
+      centuryFilter: searchParams.getAll('centuries'),
+      sexFilter: searchParams.get('sex'),
+      sortFilter: searchParams.get('sort') as SortColumns,
+      sortOrder: searchParams.get('order') as SortOrders,
+    };
+
+    return filterPeople(people, filters);
+  }, [people, searchParams]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(setPeople)
+      .catch(() => setError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
+
+  const ErrorBlock = (
+    <p data-cy="peopleLoadingError" className="has-text-danger">
+      Something went wrong
+    </p>
+  );
+
+  const EmptyPeopleBlock = (
+    <p data-cy="noPeopleMessage">
+      There are no people on the server
+    </p>
+  );
+
+  const renderingBlock = () => {
+    if (isLoading) {
+      return <Loader />;
+    }
+
+    if (error) {
+      return ErrorBlock;
+    }
+
+    if (!people.length) {
+      return EmptyPeopleBlock;
+    }
+
+    return <People people={visiblePeople} />;
+  };
```

Comment:
You also need to add a paragraph to show when there are no fitting criteria for people's table `There are no people matching the current search criteria `
![image](https://github.com/mate-academy/react_people-table-advanced/assets/96423255/e27b4f3a-a7ec-4090-b514-b18495bd7415)